test.c

```diff
@@ -1848,8 +1848,7 @@ const struct message responses[] =
   ,.http_minor= 1
   ,.status_code= 101
   ,.response_status= "Switching Protocols"
-  ,.body= "body"
-  ,.upgrade= "proto"
+  ,.upgrade= "bodyproto"
```

Comment:
See #363, it's possible someone might be relying on the current behavior.

Reply:
This PR only changes the tests for 101 responses. The HTTP_200_RESPONSE_WITH_UPGRADE_HEADER* tests are unaffected (and they all pass).

RFC 7230 explicitly forbids using Content-Length or Transfer-Encoding in 1xx responses ("A server MUST NOT send" etc.) and RFC 2616 also had similarly strict requirements ("All 1xx ... responses MUST NOT include a message-body"), so I think it would probably be ok.